pkg/repo/chartrepo.go

```diff
@@ -292,3 +292,13 @@ func (e *Entry) String() string {
 	}
 	return string(buf)
 }
+
+// URLWithTrailingSlash returns the repository URL with a trailing slash.
+// If the URL already ends with a slash, it will be returned unchanged.
+func (e *Entry) URLWithTrailingSlash() string {
+	if e.URL[len(e.URL)-1:] == "/" {
```

Comment:
This can be replaced with a call to `strings.HasSuffix(e.URL, "/")`. Less prone to array length errors and way more readable.

This could fail if `e.URL`'s length is 0.

https://golang.org/pkg/strings/#HasSuffix

You could also just replace this with `strings.TrimSuffix(e.URL, "/") + "/"`. It'll strip any trailing slash then append one to the end of the string.

Reply:
Right, I didn't have the edge case of `len(e.URL) == 0` in mind and went with your second approach now.

> It won't cover cases such as empty strings [...]

So the `TrimSuffix`-based solution will simply return `"/"` for empty strings. We could explicitly check if the passed string is empty and return an empty string if it is, but that wouldn't make a difference at least in this scenario.